Assets/MixedRealityToolkit.SDK/Experimental/Features/Input/Handlers/PrimitiveManipulator.cs

```diff
@@ -0,0 +1,744 @@
+        {
+            if (smoothingActive == false || smoothingAmount == 0)
+            {
+                return 1;
+            }
+            // Obtained from "Frame-rate independent smoothing"
+            // www.rorydriscoll.com/2016/03/07/frame-rate-independent-damping-using-lerp/
+            // We divide by max value to give the slider a bit more sensitivity.
+            return 1.0f - Mathf.Pow(smoothingAmount, Time.deltaTime);
+        }
+
+        private Vector3[] GetHandPositionArray()
+        {
+            var handPositionMap = new Vector3[pointerIdToPointerMap.Count];
+            int index = 0;
+            foreach (var item in pointerIdToPointerMap)
+            {
+                handPositionMap[index++] = item.Value.pointer.Position;
+            }
+            return handPositionMap;
+        }
+
+        public void OnFocusChanged(FocusEventData eventData)
+        {
+            bool isFar = !(eventData.Pointer is IMixedRealityNearPointer);
+            if (isFar && !AllowFarManipulation)
+            {
+                return;
+            }
+
+            if (eventData.OldFocusedObject == null ||
+                !eventData.OldFocusedObject.transform.IsChildOf(transform))
+            {
+                if (OnHoverEntered != null)
+                {
+                    OnHoverEntered.Invoke(new ManipulationEventData
+                    {
+                        ManipulationSource = gameObject,
+                        IsNearInteraction = !isFar
+                    });
+                }
+            }
+            else if (eventData.NewFocusedObject == null ||
+                    !eventData.NewFocusedObject.transform.IsChildOf(transform))
+            {
+                if (OnHoverExited != null)
+                {
+                    OnHoverExited.Invoke(new ManipulationEventData
+                    {
+                        ManipulationSource = gameObject,
+                        IsNearInteraction = !isFar
+                    });
+                }
+            }
+        }
+
+        private void ReleaseRigidBody(Vector3 velocity, Vector3 angularVelocity)
+        {
+            if (rigidBody != null)
+            {
+                rigidBody.isKinematic = wasKinematic;
+
+                if (releaseBehavior.HasFlag(ReleaseBehaviorType.KeepVelocity))
+                {
+                    rigidBody.velocity = velocity;
+                }
+
+                if (releaseBehavior.HasFlag(ReleaseBehaviorType.KeepAngularVelocity))
+                {
+                    rigidBody.angularVelocity = angularVelocity;
+                }
+            }
+        }
+
+        private PointerData GetFirstPointer()
+        {
+            // We may be able to do this without allocating memory.
+            // Moving to a method for later investigation.
+            return pointerIdToPointerMap.Values.First();
+        }
+
+        private Quaternion ApplyConstraints(Quaternion newRotation)
```

Comment:
This method actually doesn't seem so different from ManipulationHandler. Are you sure you can't just change the inspector? Seems like the only difference is that the HandMovementType and TwoHandManipulation type are flags. What else is different that requires us to have a new class?

Reply:
Yeah, there isn't a huge amount of difference yet. I separated out a lot of WIP changes so that I'm not pushing everything into one monolithic PR. I figured that it's better to separate these classes now so that we can make the changes we wanted to make in the next couple of iterations without having to worry about breaking changes.

I was going to look at constraints next, and making them consistent with BoundingBox. I already have most of the changes locally.